pkg/api/validation/validation.go

```diff
@@ -2575,6 +2575,10 @@ func ValidateServiceUpdate(service, oldService *api.Service) field.ErrorList {
 		allErrs = append(allErrs, ValidateImmutableField(service.Spec.ClusterIP, oldService.Spec.ClusterIP, field.NewPath("spec", "clusterIP"))...)
 	}
 
+	if oldService.Spec.ClusterIP == api.ClusterIPNone {
+		allErrs = append(allErrs, ValidateImmutableField(service.Spec.ClusterIP, oldService.Spec.ClusterIP, field.NewPath("spec", "clusterIP"))...)
```

Comment:
@thockin what about the case in https://github.com/kubernetes/kubernetes/blob/caad4db78be6ed/test/e2e/dns.go#L481 ? It seems like you can change ClusterIP when changing type of service. I think my check should prevail.